.circleci/config.yml

```diff
@@ -18,15 +18,24 @@ aliases:
         if [ -n "$CIRCLE_PR_NUMBER" ]; then
           .ci/setup.sh
         fi
+  - &cache_calc_key
+    run:
+      name: Calculate cache key
+      command: |
+        if [ -n "$CIRCLE_PR_NUMBER" ]; then
+          date +%Y-%m > .circleci_cachekey
+        else
+          echo not running > .circleci_cachekey
+        fi
   - &cache_restore
     restore_cache:
       keys:
-        - cache-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}-{{ .Revision }}
-        - cache-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}-
-        - cache-{{ .Environment.CIRCLE_JOB }}-
+        - cache-{{ .Environment.CIRCLE_JOB }}-{{ checksum "/usr/bin/gcc" }}-{{ checksum ".circleci_cachekey" }}-{{ .Branch }}-{{ .Revision }}
```

Comment:
Why the checksum on `gcc`?

Reply:
Last time we did a big update of the docker base image (moving from ubuntu trusty to bionic) it broke the ghc that stack had installed inside of  `~/.stack`.
My idea was to use the `gcc` binary as a proxy for big changes like that.